### PR TITLE
Implement C1 continuous version of the Erbs diffuse-fraction/decomposition model

### LINKED ISSUE #1755: Erbs diffuse fraction model has discontinuities at transition points
**Is your feature request related to a problem? Please describe.**
If you look closely at the piecewise definition of the Erbs diffuse fraction model you will see that it is discontinuous at the two transition points.  This is not usually a problem, but when the Erbs model happens to be used in some kind of numerical optimization the discontinuities could cause problems.

**Describe the solution you'd like**
Add a new diffuse fraction function with a piecewise definition that closely matches the original Erbs model but is continuous. And while we're at it we can make the first derivative continuous too.

**Describe alternatives you've considered**
Modify the existing model function.

**Additional context**
This has been implemented and tested off line but I don't have the bandwidth for a PR currently.  I thought I should mention it since it is somewhat relevant to #1753.

### PR DESCRIPTION
**Pull request by adriesse**
- [x] Closes #1755 
 - [x] I am familiar with the [contributing guidelines](https://pvlib-python.readthedocs.io/en/latest/contributing.html)
 - [x] Tests added
 - [x] Updates entries in [`docs/sphinx/source/reference`](https://github.com/pvlib/pvlib-python/blob/main/docs/sphinx/source/reference) for API changes.
 - [x] Adds description and name entries in the appropriate "what's new" file in [`docs/sphinx/source/whatsnew`](https://github.com/pvlib/pvlib-python/tree/main/docs/sphinx/source/whatsnew) for all changes. Includes link to the GitHub Issue with `` :issue:`num` `` or this Pull Request with `` :pull:`num` ``. Includes contributor name and/or GitHub username (link with `` :ghuser:`user` ``).
 - [x] New code is fully documented. Includes [numpydoc](https://numpydoc.readthedocs.io/en/latest/format.html) compliant docstrings, examples, and comments where necessary.
 - [x] Pull request is nearly complete and ready for detailed review.
 - [x] Maintainer: Appropriate GitHub Labels (including `remote-data`) and Milestone are assigned to the Pull Request and linked Issue.

### CODE REVIEW

**Comment by adriesse** [comment]
This new model/function is using the same structure as all the other diffuse-fraction/decomposition functions with one exception: it has an optional input for dni_extra, which makes datetime_or_doy also optional.

File: pvlib/irradiance.py [hunk 1]
```diff
@@ -2267,6 +2267,138 @@ def erbs(ghi, zenith, datetime_or_doy, min_cos_zenith=0.065, max_zenith=87):
...
+        If neither datetime_or_doy nor dni_extra is provided.
+
+    Notes
+    -----
+    The diffuse fraction of the Erbs-Driesse model deviates from the original
+    Erbs model by less than 0.0005.
+
+    References
+    ----------
+    .. [1] A. Driesse, A. Jensen, R. Perez, A Continuous Form of the Perez
```
Comment (cwhanse) [inline pvlib/irradiance.py:2335]: This says "Perez" model, is that a typo? Or is there also a continuous Perez model forthcoming?

**Review by cwhanse — APPROVED** [review]

File: pvlib/irradiance.py (hunk 1 above)
Comment (adriesse) [reply pvlib/irradiance.py:2335]: No. Yes.  The modification is introduced along with the Perez model.  They are used together for reverse transposition in the paper.

**Comment by adriesse** [comment]
@kandersolar you have a very good point that I have also thought about.   Here are a two options that are not mutually exclusive: 

- delay the merge and reconsider the question just before the next release
- give the PR reviewers access to our manuscript

Here is a plot from the manuscript:

erbs_discontinuity

**Comment by cwhanse** [comment]
@kandersolar we added the `sandia_multi` inverter function before the reference paper was published. The extension of `iam.physical` to handle multiple interfaces was originated in pvlib discussions, if I'm not mistaken. So we've made exceptions before.

File: pvlib/irradiance.py [hunk 2]
```diff
@@ -2267,6 +2267,136 @@ def erbs(ghi, zenith, datetime_or_doy, min_cos_zenith=0.065, max_zenith=87):
     return data
 
 
+def erbs_driesse(ghi, zenith, datetime_or_doy=None, dni_extra=None,
+                 min_cos_zenith=0.065, max_zenith=87):
+    r"""
+    Estimate DNI and DHI from GHI using the continuous Erbs-Driesse model.
+
+    The Erbs-Driesse model [1]_ is a new formulations of the original Erbs
```
Comment (wholmgren) [inline pvlib/irradiance.py:2275]: "a *new* formulation" may eventually feel not so new. "a reformulation" or something like that might be more enduring.

**Review by wholmgren — APPROVED** [review]
Thanks @adriesse! 

I don't think we need to wait for a publication in this case.

File: pvlib/irradiance.py (hunk 2 above)
Comment (adriesse) [reply pvlib/irradiance.py]: "New" vs. "original" should be enduring, but I have nevertheless reformulated the string and made a new commit.

**Comment by adriesse** [comment]
Does anyone have an idea why these test failures started to pop up?

**Comment by kandersolar** [comment]
I suspect `RuntimeError: Optimal parameters not found: The maximum number of function evaluations is exceeded.` errors are due to https://github.com/scipy/scipy/issues/19103.

**Review by kandersolar — APPROVED** [review]
Thanks @adriesse!

**Comment by adriesse** [comment]
Ok, seems ready to merge then--or do we have to wait for scipy to sort itself out?

**Comment by kandersolar** [comment]
> do we have to wait for scipy to sort itself out?

Nope, no reason to hold this up for that :)

**Comment by adriesse** [comment]
Thanks @ all!

**Pull request merged**